### ISSUE TITLE
new organization

### ISSUE THREAD

**Issue by niekdomi** [HSR-Stud/Willkommen#4]
The organizations [HSR-Stud](https://github.com/HSR-Stud) and [openhsr](https://github.com/openhsr) appear inactive. The only person contributing occasionally seems to be @NaoPross.

Since (based on my understanding) [ost-fh](https://github.com/ost-fh) is not intended for student use, I suggest creating a new organization, such as **OST-Students**.  

The current organizations, HSR-Students and openhsr, face the following issues:  

1. **Why are there two organizations?**  
   There’s a repository called "LaTeX-Workshop" on both, with only minor differences. This duplication creates confusion, and it would be clearer if only one organization existed.  

2. **Too many repositories.**  
   Each summary has its own repository. Instead, all summaries could be grouped into a single repository with organized folders. This would make navigation and maintenance much easier.  

3. **Limited awareness.**  
   Most students are likely unaware of these repositories, or they assume they’re abandoned since the school is no longer called "HSR." A new organization with a name like **OST-Students** could attract more attention within OST. With proper advertising, this could encourage students to explore and contribute to these repositories.  

What do you think about this idea? 😄

**Comment by NaoPross** [comment]
Hi there, glad to see some interest to get this thing moving again. I get your confusion, let me answer these questions before we move forward with the discussion.

> Why are there two organizations?

OpenHSR was a Verein of mostly Informatik students, while HSR-Stud was mostly (but not only) ET people and was managed by the ET fachschaft. Both went through the transition to OST but only OpenHSR registered as [openOST](https://openost.ch/) after covid, but this one was forgotten by the fachschaft (maybe lack of interest, not sure). [ost-fh](https://github.com/ost-fh) seems to be managed by a prof and a few institute researcher from informatik.

TLDR: because there are (were?) separate Verein on campus, with separate GV, etc etc.

The LaTeX workshop is something that used to be organized by ET a long time ago (HSR-Stud and [studentenportal.ch](https://studentenportal.ch) go back many years before I was a student to around ~2010), I attended to this presentation on my first semester. Later while I was a student I wanted to continue this, and ended up getting more help from the people at OpenHSR than the ET fachschaft, so the LaTeX workshop moved to them. Once I graduated I lost contact with the president of OpenHSR (he had also finished his studies) but I was asked to redo the workshop by people who had missed it the first time, so I re-organized it with the ET fachschaft where I knew some people (e.g. Flurin). To add more to the lore, the original workshop was actually LaTeX + Git, but since it was too much work I left out the Git part. Last time there seemed to be a lot of interest in Git by those who participated to the LaTeX Workshop I've been slowly preparing some [resources](https://github.com/HSR-Stud/Git-Workshop) to do a Git workshop in the spring, so that both of the pre-covid parts would be resurrected.

> This duplication creates confusion, and it would be clearer if only one organization existed.

I wanted to archive the old Open\HSR repository, but I can't, and I don't know who to ask since I'm not on campus and don't know who is the current president of Open\OST (does it still exist?).

**Comment by NaoPross** [comment]
> **Too many repositories.**
> Each summary has its own repository. Instead, all summaries could be grouped into a single repository with organized folders. This would make navigation and maintenance much easier.

This is intuitive but have you thought about the fact that with everyone working on the same monolithic repository having to do git merges would be much more common? This is not ideal for latex and git beginners (= most students). Also it would be a pretty monumental task to move all of the old stuff into this new repository, or would you start from scratch? The stuff here on HSR-Stud is not amazing but not worthless either. If you're on a time crunch for the exam sometimes there is just not enough time to create a new summary. The archive has actually a lot of value IMHO.

> **Limited awareness.**
> [...] A new organization with a name like OST-Students could attract more attention within OST.

Good idea, though not sure if that is the core of the awareness problem. I could also change the display name of the organization to OST Students, or would you be commited to help out a migration to the hypothetical new OST-Students org? 

> With proper advertising, this could encourage students to explore and contribute to these repositories.

If you're on campus feel free to create propaganda to get people interested in this. I tried to put this at the end of the LaTeX workshop but it seems like it hasn't had much effect.

**Comment by NaoPross** [comment]
Dragging in a few people that might still be on campus (but probably aren't) if they want to say something @HSR-Stud/stud

**Comment by NaoPross** [comment]
I see that @fabianhauser has registered [openOST](https://github.com/openost) but they have not migrated, and I recall the GV when the discussion about the migration happened. Do you know who is the current president at openOST?

**Comment by fabianhauser** [comment]
> Do you know who is the current president at openOST?

I'm not on campus anymore. But from what I've heard, it was dormant recently. @das-g or @the-compiler might have more infos.

> I see that @fabianhauser has registered [openOST](https://github.com/openost) but they have not migrated, and I recall the GV when the discussion about the migration happened.

The issue was that GitHub didn't allow renaming organisations with >50 members, which [openhsr](https://github.com/openhsr) has. However, maybe that has changed. I registered openOST back then so we could migrate later (if desired). I'm happy to give access to the [openost](https://github.com/openost) organisation if someone is interested in taking the lead.

**Comment by das-g** [comment]
[open\OST](https://openost.ch/) AFAIK hasn't been disbanded, thus (at least in an "on paper" sense) it does still exist. Though "dormant" might be an overstatement of its activity level, from what I could observe. Without any intention to offend, I think "comatose" would be a more fitting analogy.

**Comment by The-Compiler** [comment]
I don't know specifics, but it's certainly not working as intended. My maintenance contract to take care of keeping https://studentenportal.ch/ up and running so far wasn't renewed for 2024 (sic!), despite me reaching out in January and multiple follow-ups that apparently nobody cared about much.

**Comment by niekdomi** [comment]
First of all, thank you for your detailed reply. It is interesting to hear what was going on back then.

> This is intuitive but have you thought about the fact that with everyone working on the same monolithic repository having to do git merges would be much more common? This is not ideal for latex and git beginners.  

I believe most students who are comfortable with LaTeX also have at least a basic understanding of Git(Hub). 

> Also it would be a pretty monumental task to move all of the old stuff into this new repository.  

That’s a fair point. I’m not opposed to taking on this task if there’s significant demand for these summaries. (More on that below...)  

> A new organization with a name like OST-Students could attract more attention within OST.  

This week, I asked a few students whether they would be interested in such a repository. The feedback I received suggests there’s little interest in this idea. It seems like there isn’t much demand for such an organization, regardless of whether it’s called `HSR-Stud` or something with `OST`.  

> With proper advertising, this could encourage students to explore and contribute to these repositories.  

I can certainly ask around to gauge more interest, but as mentioned earlier, it doesn’t seem like many students are particularly interested at this point.  

If I can find a group of people genuinely interested in contributing to and maintaining such an organization, I’d be happy to revisit this topic in spring. However, for now, I’d prefer to avoid investing significant effort into something that may not be used.

**Comment by nselvara** [comment]
I'm from the 2018-2021 ET badge, so since 2021 I didn't contribute anymore. But I'd also support to move the entirety to OST-Students or something like that. This would help a lot.
I can't help, but it seems to be that there are not many students out there, who want to open source their summaries (Zusammenfassung) yet alone fix the existing ones? Or are the current ones still valid for the lectures?

**Comment by niekdomi** [comment]
I can only speak for CS since that’s my area of study. In the case of [CN1](https://github.com/HSR-Stud/CN1), there have been significant changes over the last year, so the current summary can at best serve as a template or inspiration, as it hasn’t been updated since 2013. Similarly, [EmbSW2](https://github.com/HSR-Stud/EmbSW2), which I believe has been renamed to [EmbSys2](https://studien.ost.ch/allModules/22068_M_EmbSys2.html), has been completely rewritten (or will be in the upcoming term).

I feel like many of the summaries here are legacy content that could still be used as a reference, but they are no longer recommended or used.

**Comment by NaoPross** [comment]
> I believe most students who are comfortable with LaTeX also have at least a basic understanding of Git(Hub).

I thought that too when I first joined, but from experience I can guarantee that is not the case. Well into the 3rd semester and later merge conflicts caused panics among many students. Though, CS students tend to do better than ET in this regard. In any case I hope that the git workshop will improve this state of affairs.

> I can certainly ask around to gauge more interest, but as mentioned earlier, it doesn’t seem like many students are particularly interested at this point.

This week after an exam I was too tired to do anything serious so I've made [some](https://github.com/HSR-Stud/Git-Workshop/blob/master/posters/second.pdf) [posters](https://github.com/HSR-Stud/Git-Workshop/blob/master/posters/poster.pdf) for the next event. They should be sufficiently eye-catching to get at least someone interested. Moreover, there is actually a small group of about ~5 students who are passively part of this project (one of them I think is @JoLOST1, I don't know the handles of the others). Maybe you could get in touch with them.

> If I can find a group of people genuinely interested in contributing to and maintaining such an organization, I’d be happy to revisit this topic in spring. However, for now, I’d prefer to avoid investing significant effort into something that may not be used.

That is reasonable, focus on your exams for now. It will be a project for next sprint in any case. Though I do have to say that every year there are definitely students writing new summaries, it's just that they don't bother to put them on here and work with Git (eg from my year Dubacher comes to mind), or if they do they are on other platforms (eg [GitLab](https://gitlab.com/sstitz), @gandalfgrauweiss is that you?). So the interest may have been in decline, but probably because of a lack of organization more than anything else.

**Comment by NaoPross** [comment]
> My maintenance contract to take care of keeping https://studentenportal.ch/ up and running so far wasn't renewed for 2024 (sic!)

That is a shame, because it seems to be used by many, though I suspect most users don't know who is actually behind the whole thing. I hope that next semester things will start moving again. I'll try to publicize it a bit at the next event.

@domi413 in January I might come on campus to study for my exams a few times so we could have a chat on how to best move forward. Or if you're too busy with your exams I'll also be there later (TBD) in spring for the git workshop.

**Comment by niekdomi** [comment]
It will probably be spring. I hadn't planned to study at OST due to the travel time.

**Comment by nselvara** [comment]
Maybe, the Fachschaft Elektrotechnik (if it still exists 🙃) should try to promote the platform that everyone might benefit by sharing their summaries.

**Comment by NaoPross** [comment]
@domi413 I am done with my exams, and I started to organize the Git workshop, where we will publicise this Org. If you want you can send me an email to np (at) 0hm.ch with an address / phone number to keep you in the loop. Also, I'm on campus right now.

**Comment by flurin-b** [comment]
> Maybe, the Fachschaft Elektrotechnik (if it still exists 🙃) should try to promote the platform that everyone might benefit by sharing their summaries.

It does still exist, although it is mostly concerned with organizing student events like E-Abig, bars, glüewii and the christmas bar recently.  We do use the fachschaft-channel to promote the LaTeX and Git workshops, maybe we could do better with pushing for engagement on here though.

Recently, @sstitz and @P4ntomime (on github) have created great summaries for most courses as well as a truly beautiful summary template. Some of them were even updated by some first semesters last semester. I don't know if any of them ever made it on here or if they just reside publicly on their personal repos though. If they're not on here, moving them here might bring some traffic too.

A lot of E summaries are also just PDFs [on the Fachschaft E Google drive](https://drive.google.com/drive/u/0/folders/0B_hSK399-tZrUVJGemJWSWtYdTA?resourcekey=0-hkgMfeL6bx5EPVUVI2Ar6A) (which I'm currently "managing") so lots of students just grab what they need from there and don't even realize this repo exists.
It might be smart to just nuke the drive and make sure we move over all the summaries and make all the summary repos either auto build a pdf or at least include them so people can just download them and don't need to build them themselves (since a lot of students lack the skills). This would be a LOT of work though... Also there's a lot of old exams on there which we would need to put somewhere.


All in all I think if the LaTeX and Git workshops keep happening and we keep promoting this repo we might find some people that are interested in keeping it going.
Maybe we can do more promotion via the fachschaft, although we do already tell all of the students about this repo on their first day.
Finally, nuking the google drive to force students on here would be an option, it would just take a lot of work. Also some people who are uploading things there would probably just give up since they have no idea how to use Git or GitHub.

**Comment by The-Compiler** [comment]
*edit: I forgot that I already said some of this [above](https://github.com/OST-Stud/Willkommen/issues/4#issuecomment-2556478740) last December. Whoops! Well, at least there is some additional context here.*

For what it's worth, there's also the [Studentenportal](https://studentenportal.ch/) which is (/was until recently) run by me via open\\OST with a lot of previous summaries, especially for Informatik modules, but not only.

Unfortunately it's somewhat dead at this point, also from a maintenance side of view. I used to take care of server/software maintenance and user support for it via my company (Bruhin Software) with a yearly contract with open\\(HSR|OST), but unfortunately any contact seems to have ceased since early 2024 (sic!). This approach has worked out fine for 2020-2023, after nobody was to be found wanting to take care of the project in their free time (me included as I already have two major open source projects on my hands).

There used to be some ideas around integrating with existing efforts like the ET git repos:

- https://github.com/studentenportal/web/issues/66
- https://github.com/studentenportal/web/issues/183
- https://github.com/studentenportal/web/issues/256

but then again right now there are probably more pressing matters (like upgrading to a supported Django and Ubuntu/Debian version again) which I'm unable to tackle alone as an unpaid side-project.

If someone wants to help and can seriously put in a commitment for taking care of it for a while (or try to get a community effort going again); or can restore contact to open\\OST to possibly do this as a paid effort again, I'd definitely be interested in helping with getting things up to speed again.
